nifi-toolkit/nifi-toolkit-encrypt-config/src/main/groovy/org/apache/nifi/toolkit/encryptconfig/DecryptMode.groovy

```diff
@@ -0,0 +1,322 @@
+                break
+
+            default:
+                throw new RuntimeException("Unsupported file type '${config.fileType}'")
+        }
+
+        if (!decryptedSerializedContent) {
+            throw new RuntimeException("Failed to load and decrypt input file.")
+        }
+
+        if (config.outputToFile) {
+            try {
+                File outputFile = new File(config.outputFilePath)
+                if (ToolUtilities.isSafeToWrite(outputFile)) {
+                    outputFile.text = decryptedSerializedContent
+                    logger.info("Wrote decrypted file contents to '${config.outputFilePath}'")
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("Encountered an exception writing the decrypted content to '${config.outputFilePath}': ${e.getMessage()}", e)
+            }
+        } else {
+            System.out.println(decryptedSerializedContent)
+        }
+
+    }
+
+    private CliBuilder cliBuilder() {
+
+        String usage = "${EncryptConfigMain.class.getCanonicalName()} decrypt [options] file"
+
+        int formatWidth = EncryptConfigMain.HELP_FORMAT_WIDTH
+        HelpFormatter formatter = new HelpFormatter()
+        formatter.setWidth(formatWidth)
+        formatter.setOptionComparator(null) // preserve order of options below in help text
+
+        CliBuilder cli = new CliBuilder(
+                usage: usage,
+                width: formatWidth,
+                formatter: formatter,
+                stopAtNonOption: false)
+
+        cli.h(longOpt: 'help', 'Show usage information (this message)')
+        cli.v(longOpt: 'verbose', 'Enables verbose mode (off by default)')
+
+        // Options for the password or key or bootstrap.conf
+        cli.p(longOpt: 'password',
+                args: 1,
+                argName: 'password',
+                optionalArg: true,
+                'Use a password to derive the key to decrypt the input file. If an argument is not provided to this flag, interactive mode will be triggered to prompt the user to enter the password.')
+        cli.k(longOpt: 'key',
+                args: 1,
+                argName: 'keyhex',
+                optionalArg: true,
+                'Use a raw hexadecimal key to decrypt the input file. If an argument is not provided to this flag, interactive mode will be triggered to prompt the user to enter the key.')
+        cli.b(longOpt: 'bootstrapConf',
+                args: 1,
+                argName: 'file',
+                'Use a bootstrap.conf file containing the master key to decrypt the input file (as an alternative to -p or -k)')
+
+        cli.o(longOpt: 'output',
+                args: 1,
+                argName: 'file',
+                'Specify an output file. If omitted, Standard Out is used. Output file can be set to the input file to decrypt the file in-place.')
+
+        return cli
+
+    }
+
+    static class DecryptConfiguration {
```

Comment:
Should this implement the `Configuration` interface?

Reply:
Yes.
- [ ] Adopt the Configuration interface where applicable to convey intended class structure to future developers as the Configuration interface is filled in more.